src/manifolds/StiefelSubmersionMetric.jl

```diff
@@ -0,0 +1,290 @@
+@doc raw"""
+    StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+
+The submersion (or normal) metric family on the [`Stiefel`](@ref) manifold.
+
+# Constructor
+
+    StiefelSubmersionMetric(α)
+
+Construct the submersion metric on the Stiefel manifold with the parameter ``α > -1``.
+
+The submersion metric family has two special cases:
+- ``α = -\frac{1}{2}``: [`EuclideanMetric`](@ref)
+- ``α = 0``: [`CanonicalMetric`](@ref)
+"""
+struct StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
```

Comment:
Hm, now that I though about this, the Stiefel is already mentioned in a manifold whenever this is used so maybe SubmersionMetric alone would be fine, too?

Reply:
I don't think I like that. e.g. for `CanonicalMetric` we created `ApproximateLogarithmicMap`, which is a very generic name, but the implemented fields are just those needed for `CanonicalMetric`, and that object won't necessarily be useful for every approximate logarithmic map one might need. Since we have no use for a submersion metric outside of `Stiefel`, I lean towards keeping it specific.

Reply:
I have two points why I would prefer a generic name

* using the metric we will always use it with Stiefel so in any signature the name Stiefel doubles
* if we have a use for a submersion metric, changing the name here would be a breaking change.

But you are also right, with the alpha it might be quite specific for Stiefel; then the second point is obsolete – still in any function signature it reads a little strange with the Stiefel appearing twice.

Reply:
I also prefer the name `StiefelSubmersionMetric`.